dcm4che-core/src/main/java/org/dcm4che3/io/DicomInputStream.java

```diff
@@ -703,23 +700,25 @@ private void readSequence(int len, Attributes attrs, int sqtag)
         String privateCreator = attrs.getPrivateCreator(sqtag);
         boolean undefLen = len == -1;
         long endPos = pos + (len & 0xffffffffL);
-        for (int i = 0; undefLen || pos < endPos; ++i) {
-            readHeader();
-            if (tag == Tag.Item) {
-                handler.readValue(this, seq);
-            } else if (tag == Tag.SequenceDelimitationItem) {
-                if (length != 0)
-                    skipAttribute(UNEXPECTED_NON_ZERO_ITEM_LENGTH);
-                break;
-            } else
-                skipAttribute(UNEXPECTED_ATTRIBUTE);
+        for (int i = 0; (undefLen || pos < endPos) && readItemHeader(); ++i) {
+            addItemPointer(sqtag, privateCreator, i);
```

Comment:
And this set of changes here preserve the pointer data is what makes the change work